ra/ra.go

```diff
@@ -203,27 +196,28 @@ func NewRegistrationAuthorityImpl(
 	}, []string{"reason"})
 	stats.MustRegister(revocationReasonCounter)
 
-	authzAges := prometheus.NewHistogram(prometheus.HistogramOpts{
+	authzAges := prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name: "authz_ages",
-		Help: "Histogram of ages of Authorization objects when they're attached to a new Order",
-		// authzAges keeps track of how old, in seconds, authorizations were at
-		// the time that we attached them to an order. We give it a non-standard
-		// bucket distribution so that the leftmost (closest to zero) bucket can be
-		// used exclusively for brand-new (i.e. not reused) authzs. Our buckets are:
-		// one nanosecond, one second, one minute, one hour, 7 hours (our CAA reuse
-		// time), 1 day, 2 days, 7 days, 30 days, +inf (should be empty).
+		Help: "Histogram of ages, in seconds, of Authorization objects, labelled by method and type",
+		// authzAges keeps track of how old, in seconds, authorizations are when
+		// we attach them to a new order and again when we finalize that order.
+		// We give it a non-standard bucket distribution so that the leftmost
+		// (closest to zero) bucket can be used exclusively for brand-new (i.e.
+		// not reused) authzs. Our buckets are: one nanosecond, one second, one
+		// minute, one hour, 7 hours (our CAA reuse time), 1 day, 2 days, 7
+		// days, 30 days, +inf (should be empty).
 		Buckets: []float64{0.000000001, 1, 60, 3600, 25200, 86400, 172800, 604800, 2592000, 7776000},
-	})
+	}, []string{"method", "type"})
 	stats.MustRegister(authzAges)
 
-	orderAges := prometheus.NewHistogram(prometheus.HistogramOpts{
+	orderAges := prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name: "order_ages",
-		Help: "Histogram of ages of Order objects when they're Finalized",
+		Help: "Histogram of ages, in seconds, of Order objects when they're reused and finalized, labelled by method",
 		// Orders currently have a max age of 7 days (168hrs), so our buckets are:
 		// 1 second, 10 seconds, 1 minute, 10 minutes, 1 hour, 10 hours, 1 day,
 		// 7 days, +inf.
 		Buckets: []float64{1, 10, 60, 600, 3600, 36000, 86400, 172800},
```

Comment:
Any desire to add the 1e-9 (i.e. 0) bucket, like we have for authz_ages?

Reply:
Sure, it would be useful to disambiguate between orders retrieved and truly new orders.